src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs

```diff
@@ -196,10 +196,7 @@ static MonitoredEndpoint[] GetMonitoredEndpoints(EndpointRegistry endpointRegist
         static MonitoredEndpointMessageType[] GetMonitoredMessageTypes(MessageTypeRegistry registry, string endpointName)
         {
             return registry.GetForEndpointName(endpointName)
-                .Select(mt => new MonitoredEndpointMessageType
-                {
-                    MessageType = mt.MessageType
-                })
+                .Select(mt => MonitoredEndpointMessageType.FromTypeName(mt.MessageType))
```

Comment:
Could this be moved to `MessageTypeRegistry`? It would run transformation only once for each message type. No need to do the computation on every access.

Reply:
I'm not sure `MessageTypeRegistry` is a proper place. `MonitoredEndpointMessageType` is part of api contract and we should not add dependency to it from infrastructural parts. 

I do think that we should cache the values though. I'll create a dedicated caching parser for this purpose.